network.go

```diff
@@ -181,6 +181,19 @@ func (s *Server) setupTLS() (err error) {
 		return err
 	}
 
+	tlsc.CipherSuites = []uint16{
+		tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+		tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+		tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+		tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+		tls.TLS_RSA_WITH_AES_128_CBC_SHA256,
+	}
+
+	tlsc.PreferServerCipherSuites = true
+	tlsc.MinVersion = tls.VersionTLS12
+	tlsc.CurvePreferences = []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256}
```

Comment:
I am fine with configuring like this though I know nada about whats good and bad.

Question about Go's TLS_CHACHA20_POLY1305_SHA256 what do you think about that? It's apparently pretty quick between 2 go servers if you have that

We'll need an issue for this ietc

Reply:
Will need to double check this.  Fwiw, setting up nats like this bypasses some settings they tried to setup, it looks like.

Reply:
Looks like that's coming with tls1.3, so should enable it.

Reply:
I can add a lookup table fwiw, but it's still be up to us to support :(. Considering submitting a PR to golang.

Reply:
I am ok with the project making informed decisions and leaving it not configurable until someone asks :)

If you can find out about that ChaCha20 thing and I say lets get this in, we might also propose same PR to NATS I guess?

Reply:
Shouldnt we just go and set it here https://github.com/choria-io/go-security/blob/5b821bd3dbaefd3c375bdf850fd2db52aa41ff8a/filesec/file_security.go#L589 then all things get the benefit?

Reply:
We can probably drop the SHA384 ones? they are slow and I don't want to end up defaulting to some terrible thing on large networks. Had we set nothing go would go and order them based on CPU capabilities by default, but thats not what we're doing.

Reply:
I'll make an interim list of ciphersuites like https://github.com/golang/go/commit/0ee22d97ad1d1d40df1a58b0efae022e4f82f1b5 and add that to the security provider, then make it user configurable.  When 1.14 drops we can remove the list we need to maintain.